### ISSUE TITLE
Allow loop in FromDump

### ISSUE THREAD

**Issue by Qcloud1223** [tbarbette/fastclick#386]
In commit 5a92237b, `FromIPSummaryDump` adds another argument `TIME` to allow specifying loop time (the Wiki is not up-to-date though). This is very useful when implementing a packet generator.

I'm a little suprised to see such feature still unsupported in `FromDump`. Is there any reason preventing FastClick from doing this?

Thanks,
Yihan

**Comment by tbarbette** [comment]
Great you found it useful !

There is no particular reason. Though, each new feature add a bit of slowiness. It can be compensated by PacketMill but for large pipeline PacketMill might need adjustement in some replacements.

What we have done though is to use Replay and mostly ReplayUnqueue to read all packets from FromDump, keep them in memory and replay them X times. With TIMING 100 it will keep the original timing. https://github.com/tbarbette/fastclick/blob/main/conf/pktgen/pktgen-l2-mt-replaypcap.click uses this element with FromDump but without the TIMING.

The need for TIME directly in FromIPSummaryDump is for hosts where memory is too limited and we can't really preload, that's why the ReplayUnqueue solution is not sufficient there.

**Comment by tbarbette** [comment]
For the documentation it's built from https://github.com/tbarbette/fastclick/blob/main/elements/analysis/fromipsumdump.hh . Apparently we did not write the documentation at that point...

**Comment by Qcloud1223** [comment]
Thank you very much! `pktgen-l2-mt-replaypcap` is exactly what I am looking for. 

I also found that `playpcap` (the version without `re`) does not implement looping (though `$replay_count is actually defined`). I guess it's because for large enough capture files that cannot be preloaded, playing them once would take long enough, so there's no point to rewind (also it deviates from `play` if trying to rewind).

`ReplayUnqueue` is also a very useful and feature-rich element. Though again I had to dive into the code to find out what `TIMING` actually means: `0` is to replay as fast as possible, and `100` is to respect the original timestamp on packets. Maybe it will be helpful to add some brief description on that one? I think it will greatly help FastClick beginners like me :)
